### ISSUE TITLE
Build fails on `4.12.0` if project folder name contains `ember-data-`

### ISSUE THREAD

**Issue by dagroe** [emberjs/data#8606]
Build fails on `4.12.0`

### Reproduction

- create new ember project: `ember new ember-data-bug --lang en` using `ember-cli: 4.12.1` `node: 18.13.0` 
- upgrade `ember-data` to `4.12.0`
- build fails with `[NodeWrapper:8 /Users/myuser/ember-data-bug/tests] is not a SourceNode`

I also tried to upgrade all dependencies to latest and pin them in my reproduction here: https://github.com/dagroe/ed-bug-reproduction

### Description
I tried a bunch of different combinations, e.g. plain new ember install, upgrade all to latest, upgrade everything to latest but set `ember-source: 4.12.0`. Tried `yarn` and `npm` installs. 

Everything works fine on `4.11.3` but fails with the error above on `4.12.0`

### Versions

```cli
└─ ember-data-bug@workspace:.
   └─ ember-source@npm:4.12.0 [a2ed2] (via npm:4.12.0 [a2ed2])
└─ ember-data-bug@workspace:.
   └─ ember-cli@npm:4.12.1 (via npm:4.12.1)
└─ ember-data-bug@workspace:.
   └─ ember-data@npm:4.12.0 [a2ed2] (via npm:4.12.0 [a2ed2])
```

**Comment by runspired** [comment]
As far as I can tell builds are working with 4.12

**Comment by runspired** [comment]
Closing because even the reproduction app works when I run `yarn` to install and `ember build` to build or `ember serve` to serve

**Comment by dagroe** [comment]
Very strange. Indeed, when I clone the repo it runs fine.

Drove me almost crazy, since the exact same project didn't run in my original folder where I created it. Finally I found that the error occurs when the project folder includes the string `ember-data`. My original project was `ember-data-bug` but I pushed it to github as `ed-bug-reproduction`. When I clone the `ed-bug-reproduction` repo into something like `ember-data-test` I get the error described above.

**Comment by gilest** [comment]
I have also (accidentally) encountered this

**Comment by runspired** [comment]
@gilest yeah its honestly one of the most curious bugs I've heard in a long time and ordinarily I'd jump at investigating but haven't had time. Would love a report/fix if you can dig around, its... WAT

**Comment by gilest** [comment]
Not sure I'll get all the way to a fix but...

- Occurs when running a development or test server (with livereload). NOT when running `ember test` (no server)
- Occurs with both classic and Embroider builds

**Comment by bertdeblock** [comment]
Running into the same issue. My addon's name is `@bagaaravel/ember-data-extensions`.
On my machine the folder name was `ember-data-extensions`. Renaming the folder "works".

**Comment by runspired** [comment]
@bertdeblock think you could dig into the broccoli side a bit?

**Comment by runspired** [comment]
The only EmberData code I could imagine *might* interplay to result in this has existed a long time. It's this check:

https://github.com/emberjs/data/blob/4251fd7105ea9a8eae17efb4e3bf8bb795b42ca4/packages/private-build-infra/src/addon-build-config-for-data-package.js#L118C5-L127C7

but reading that I don't see how it could affect.

**Comment by bertdeblock** [comment]
@runspired I've been looking into it a bit, but haven't found anything yet. Might continue looking at it today.

**Comment by runspired** [comment]
anyone know if this still happens in 4.13 or 5.3+ ? I'm guessing the shift to v2 addons eliminated whatever the issue was here but its not clear

**Comment by dagroe** [comment]
@runspired I tried a fresh ember quickstart app with `4.13.0-alpha.5` and `5.3.11`, both work fine. I guess this can be closed.
